mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -233,7 +233,7 @@ private class InternalKMeansModelWriter extends MLWriterFormat with MLFormatRegi
         ClusterData(idx, center)
     }
     val dataPath = new Path(path, "data").toString
-    sparkSession.createDataFrame(data.toImmutableArraySeq).repartition(1).write.parquet(dataPath)
+    ReadWriteUtils.saveArray(dataPath, data, sparkSession)
```

Comment:
Cann't spark.write support local filesystem path?

Reply:
I don't think so.

Reply:
if we use `spark_dataframe.write.save("file:///....")`, then it result in writing data to remote worker local disk.

Reply:
no, unless we can force the query to run on the driver.

Reply:
Are we talking about writing the data in Spark Connect client side?

Reply:
If that's the case, yeah there's no way to write it by using SparkSession.